pkg/build/controller/common/util.go

```diff
@@ -42,6 +47,75 @@ func HandleBuildCompletion(build *buildapi.Build, runPolicies []policy.RunPolicy
 	if err := runPolicy.OnComplete(build); err != nil {
 		utilruntime.HandleError(fmt.Errorf("failed to run policy on completed build %s/%s: %v", build.Namespace, build.Name, err))
 	}
+	if err := HandleBuildPruning(buildutil.ConfigNameForBuild(build), build.Namespace, buildLister, buildConfigGetter, buildDeleter); err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to prune old builds %s/%s: %v", build.Namespace, build.Name, err))
+	}
+}
+
+type ByCreationTimestamp []buildapi.Build
+
+func (b ByCreationTimestamp) Len() int {
+	return len(b)
+}
+
+func (b ByCreationTimestamp) Swap(i, j int) {
+	b[i], b[j] = b[j], b[i]
+}
+
+func (b ByCreationTimestamp) Less(i, j int) bool {
+	return !b[j].CreationTimestamp.Time.After(b[i].CreationTimestamp.Time)
+}
+
+// HandleBuildPruning handles the deletion of old successful and failed builds
+// based on settings in the BuildConfig.
+func HandleBuildPruning(buildConfigName string, namespace string, buildLister buildclient.BuildLister, buildConfigGetter buildclient.BuildConfigGetter, buildDeleter buildclient.BuildDeleter) error {
+	buildConfig, err := buildConfigGetter.Get(namespace, buildConfigName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	var buildsToDelete []buildapi.Build
+	var errList []error
+
+	if buildConfig.Spec.SuccessfulBuildsHistoryLimit != nil {
+		successfulBuilds, err := buildutil.BuildConfigBuilds(buildLister, namespace, buildConfigName, func(build buildapi.Build) bool { return build.Status.Phase == buildapi.BuildPhaseComplete })
+		if err != nil {
+			return err
+		}
+		sort.Sort(ByCreationTimestamp(successfulBuilds.Items))
+
+		successfulBuildsHistoryLimit := int(*buildConfig.Spec.SuccessfulBuildsHistoryLimit)
+		glog.V(4).Infof("Preparing to prune %v old successful builds.", (len(successfulBuilds.Items) - successfulBuildsHistoryLimit))
```

Comment:
move this log into the if block, and/or log more details:  number of builds found, number of builds configured to keep, number that will be pruned.